src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj

```diff
@@ -12,6 +12,66 @@
     <DefineConstants>$(DefineConstants);CORECLR</DefineConstants>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <StronglyTypedOutputFolder>gen</StronglyTypedOutputFolder>
+    <SkipWPFcsproj>$([System.String]::Copy('$(MSBuildProjectFile)').Contains('wpftmp'))</SkipWPFcsproj>
+  </PropertyGroup>
+
+  <Target Name = "CreateDirectories" BeforeTargets = "PSResources">
+    <MakeDir
+       Directories = "$(StronglyTypedOutputFolder)"/>
+  </Target>
+
+  <ItemGroup>
+    <!-- We have to use the workaround for the WPF project.
+        .Net 5 Preview.2 has some issues with WPF projects:
+          - Strong typed resources does not work as expected out of box.
+            (Really they are not generated at all.)
+            So we assign items in 'PSEmbeddedResource' to expected format
+            and explicitly call 'GenerateResource' task from custom 'PSResources' target.
+          - .Net 5 Preview.2 generates a temporory csproj file
+            that duplicates call our custom 'PSResources' target.
+            To avoid this we skip a project with 'wpftmp' in name.
+    -->
+    <PSEmbeddedResource Include = "resources/*.resx" Condition = "'$(SkipWPFcsproj)' == 'false'">
+      <Generator>ResXFileCodeGenerator</Generator>
+      <StronglyTypedLanguage>CSharp</StronglyTypedLanguage>
+
+      <PublicClass>$([System.String]::new('%(Filename)').StartsWith('Public', System.StringComparison.OrdinalIgnoreCase))</PublicClass>
+
+      <StronglyTypedManifestPrefix>$(RootNamespace).resources</StronglyTypedManifestPrefix>
+      <StronglyTypedManifestPrefix Condition = "'%(PublicClass)' == 'true'">$(RootNamespace).resources.public</StronglyTypedManifestPrefix>
+
+      <StronglyTypedFileName>$(StronglyTypedOutputFolder)/%(Filename).cs</StronglyTypedFileName>
+      <StronglyTypedFileName Condition = "'%(PublicClass)' == 'true'">$(StronglyTypedOutputFolder)/$([System.String]::Copy('%(Filename)').Substring(7)).cs</StronglyTypedFileName>
+
+      <StronglyTypedClassName>%(Filename)</StronglyTypedClassName>
+      <StronglyTypedClassName Condition = "'%(PublicClass)' == 'true'">$([System.String]::Copy('%(Filename)').Substring(7))</StronglyTypedClassName>
+
+      <ManifestResourceName>$(RootNamespace).resources.%(PSEmbeddedResource.StronglyTypedClassName)</ManifestResourceName>
+      <ManifestResourceName Condition = "'%(PublicClass)' == 'true'">$(RootNamespace).resources.public.%(PSEmbeddedResource.StronglyTypedClassName)</ManifestResourceName>
+    </PSEmbeddedResource>
+  </ItemGroup>
+
+  <Target Name="PSResources" Condition = "'$(SkipWPFcsproj)' == 'false'">
```

Comment:
I believe you should be declaring your Target's Inputs and Outputs, similar to Arcade's resource generator.

https://github.com/dotnet/arcade/blob/d6baec6b932970d1ef47238db8288b9da6e2c9ff/src/Microsoft.DotNet.Arcade.Sdk/tools/GenerateResxSource.targets#L45-L59